components/channel_header/channel_header.tsx

```diff
@@ -201,16 +213,19 @@ class ChannelHeader extends React.PureComponent {
         actions.openModal(modalData);
     }
 
-    showChannelHeaderPopover = (headerText) => {
-        const headerDescriptionRect = this.headerDescriptionRef.current.getBoundingClientRect();
-        const headerPopoverTextMeasurerRect = this.headerPopoverTextMeasurerRef.current.getBoundingClientRect();
+    showChannelHeaderPopover = (headerText: string) => {
+        const headerDescriptionRect = this.headerDescriptionRef.current?.getBoundingClientRect();
+        const headerPopoverTextMeasurerRect = this.headerPopoverTextMeasurerRef.current?.getBoundingClientRect();
         const announcementBarSize = 40;
-        if (headerPopoverTextMeasurerRect.width > headerDescriptionRect.width || headerText.match(/\n{2,}/g)) {
-            this.setState({showChannelHeaderPopover: true, leftOffset: this.headerDescriptionRef.current.offsetLeft});
+
+        if (headerPopoverTextMeasurerRect && headerDescriptionRect) {
+            if (headerPopoverTextMeasurerRect.width > headerDescriptionRect.width || headerText.match(/\n{2,}/g)) {
+                this.setState({showChannelHeaderPopover: true, leftOffset: this.headerDescriptionRef.current?.offsetLeft || 0});
+            }
         }
 
         // add 40px to take the global header into account
-        const topOffset = (announcementBarSize * this.props.announcementBarCount) + 40;
+        const topOffset = (announcementBarSize * (this.props.announcementBarCount || 1)) + 40;
```

Comment:
As a fallback, I believe `1` is a better option to keep `announcementBarSize` size. 
In the [comment](https://github.com/mattermost/mattermost-webapp/pull/9276#discussion_r741018457) for another PR, it was suggested to use `0`. Please let me know which one we should use?

Reply:
I don't think this fallback is actually needed since the value that comes from the state is always defined. I think it'd be better to just change the type to not be nullable instead of adding this

Reply:
You are right. It is always defined in the state, so I changed type to be non-nullable.